extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/TestScenarioStepType.java

```diff
@@ -0,0 +1,27 @@
+package com.azure.autorest.extension.base.model.codemodel;
+
+/**
+ */
+public enum TestScenarioStepType {
+
+    /**
+     * Step to run a swagger operation defined rest call. This may not be just one http call.
+     *
+     * If the operation is a long running operation (LRO), then follow the LRO polling strategy.
+     * Response statusCode must be 200 if the LRO succeeded, no matter what code the initial response is.
+     * If the LRO is PUT/PATCH, the runner should automatically insert a GET after the polling to verify the resource update result.
+     * If the operation is DELETE, then after the operation, the runner should automatically insert a GET to verify resource cannot be found.
+     * Rest call step could be defined either by an example file, or by resourceName tracking and update.
+     */
+    restCall,
+    /**
+     * Step to deploy ARM template to the scope.
+     */
+    stepArmTemplate,
+    /**
+     * Step to deploy ARM deployment script to the scope.
+     */
+    stepArmDeploymentScript
```

Comment:
Is it possible to use standard Java enum patten e.g. `REST_CALL`?

Reply:
I'll check snakeyaml documentation.

Reply:
You can try, if hard, I can live with it (if it does not appear lot on code).

But I probably not able to live with `set_resolvedSteps` :-(
Last solution would be here https://github.com/Azure/autorest.java/blob/main/extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/CodeModelCustomConstructor.java#L214-L221

Reply:
Yes, I followed the code https://github.com/Azure/autorest.java/blob/main/extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/CodeModelCustomConstructor.java#L31-L58 and can register my own enum customization.

For `_resolvedSteps` I'm wondering if I could make a custom annotation mimicking `JsonProperty` to customize the mapping between yaml and pojo. Do you think it's viable? Or I'll just hardcode my property name like in the example.

Reply:
For enum, it seems you do not need to customize CodeModelCustomConstructor?
https://github.com/Azure/autorest.java/blob/main/extension-base/src/main/java/com/azure/autorest/extension/base/model/codemodel/RequestParameterLocation.java only use the value as the in yaml, but enum name is Java.

If snakeyaml support annotation to handle the mapping, it should be good to go.
If not, you might need to go CodeModelCustomConstructor this time.

Reply:
I'm afraid I have to do the manual mapping to handle specific enum types, otherwise same error will be thrown...
>  org.yaml.snakeyaml.error.YAMLException: Unable to find enum value 'restCall' for enum class: com.azure.autorest.extension.base.model.codemodel.TestScenarioStepType

Reply:
Does e.g. `RestCall` works?

Reply:
OK I see the code
```
            } else if (type.equals(RequestParameterLocation.class)) {
                return RequestParameterLocation.fromValue(((ScalarNode) node).getValue());
            } else if (type.equals(SerializationStyle.class)) {
                return SerializationStyle.fromValue(((ScalarNode) node).getValue());
            } else if (type.equals(KnownMediaType.class)) {
                return KnownMediaType.fromValue(((ScalarNode) node).getValue());
```

Reply:
Sure, you need to do this in CodeModelCustomConstructor

Reply:
ok